aptos-move/move-examples/shared_account/sources/shared_account.move

```diff
@@ -0,0 +1,136 @@
+// This module demonstrates a basic shared account that could be used for NFT royalties
+// Users can (1) create a shared account (2) disperse the coins to multiple creators
+module shared_account::SharedAccount {
+    use std::errors;
+    use std::signer;
+    use std::vector;
+    use AptosFramework::Account;
+    use AptosFramework::Coin;
+
+    // struct Share records the address of the share_holder and the numerator and the denominator of a fraction
+    struct Share has store {
+        share_holder: address,
+        numerator: u64,
+    }
+
+    // Resource representing a shared account
+    struct SharedAccount has key {
+        share_record: vector<Share>,
+        total_shares: u64,
+        signer_capability: Account::SignerCapability,
+    }
+
+    struct SharedAccountEvent has key {
+        resource_addr: address,
+    }
+
+    const EACCOUNT_NOT_FOUND: u64 = 0;
+    const ERESOURCE_DNE: u64 = 1;
+    const EINSUFFICIENT_BALANCE: u64 = 2;
+
+    // Create and initialize a shared account
+    public entry fun initialize(source: &signer, seed: vector<u8>, addresses: vector<address>, numerators: vector<u64>) {
+        let i = 0;
+        let total = 0;
+        let share_record = vector::empty<Share>();
+
+        while (i < vector::length(&addresses)) {
+            let num_shares = *vector::borrow(&numerators, i);
+            let addr = *vector::borrow(&addresses, i);
+            assert!(Account::exists_at(addr), errors::invalid_argument(EACCOUNT_NOT_FOUND));
+
+            vector::push_back(&mut share_record, Share { share_holder: addr, numerator: num_shares });
+            total = total + num_shares;
+            i = i + 1;
+        };
+
+        let (resource_signer, resource_signer_cap) = Account::create_resource_account(source, seed);
+        move_to(
+            &resource_signer,
+            SharedAccount {
+                share_record,
+                total_shares: total,
+                signer_capability: resource_signer_cap,
```

Comment:
resource_signer_cap doesn't need to be used later. Do we store it just because it can't be dropped?

Reply:
I think we're using this on line 70 to create the resource signer: 
`let resource_signer = Account::create_signer_with_capability(&shared_account.signer_capability);`

Reply:
Ah cool makes sense! I missed it earlier